bundles/org.openhab.binding.webthing/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="webthing"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<!-- Sample Thing Type -->
+	<thing-type id="generic">
+		<label>WebThing Binding Thing</label>
+		<description>Sample thing for WebThing Binding</description>
+
+		<config-description>
+			<parameter name="webThingURI" type="text" required="true">
+				<label>webThingURI</label>
+				<description>This is the url of the webthing </description>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+
+	<channel-type id="number">
+		<item-type>Number</item-type>
+		<label>Webthing Binding Channel</label>
+		<description>Number channel for Webthing Binding</description>
+	</channel-type>
```

Comment:
I think you don't need to define the Channel types here if they are created fully dynamically.

Reply:
I have tried this without the channel description. Defining items referring to the auto-created channels worked.

Unfortunately, the oh3 user interface does not detect/show the channels. Should this work? Is this a bug in the new oh3 user interface? I would prefer to remove this channel definition. It feels like a workaround

Reply:
> 
> 
> I have tried this without the channel description. Defining items referring to the auto-created channels worked.

Without the description or without the whole `channel-type` tag? Did you try remove the entire tag?

> Unfortunately, the oh3 user interface does not detect/show the channels. Should this work? Is this a bug in the new oh3 user interface? I would prefer to remove this channel definition. It feels like a workaround

The Channels will only be shown if you add the corresponding `channel` tags in the `thing-type` tag or by adding them dynamically.

Reply:
I implemented the code to add channel dynamically according to https://www.openhab.org/docs/developer/bindings/#updating-the-thing-structure Unfortunately, the Channels will only be shown without the channel types entries in the thing-types.xml

Reply:
I have updated my tests to use Junit5

Reply:
So how do we proceed with topic? Essentially, the current solution works

Reply:
I missed that you need an existing channel type when creating a Channel. It's all fine.